cmd/thanos/compact.go

```diff
@@ -147,6 +148,9 @@ func runCompact(
 
 	ctx, cancel := context.WithCancel(context.Background())
 	f := func() error {
+		if err := os.RemoveAll(downsamplingDir); err != nil {
```

Comment:
lets move this up to line `136` and change it to `os.RemoveAll(dataDir)` and we are good to go